### PR TITLE
added apriltag_detector

### PR DESCRIPTION
**Pull request by berndpfrommer**
# Please add apriltag_detector to be indexed in the rosdistro.
ROS distros: humble, iron, rolling

Note: there is already a [apriltag_ros](https://github.com/christianrauch/apriltag_ros) package, so why add another one?
The existing apriltag_ros package not only detects apriltags, but also uses them to compute the pose of the tags (transforms). For that reason it subscribes to calibration (camera_info) messages which are *not available* until the camera has been calibrated. However, this makes the existing  package difficult to use for camera calibration.
The new package proposed here focuses only on the elementary task of perception: detect the tag's ID and corners in the image. Note that it uses the same message format as the apriltag_ros package, except that it does not utilize the homography field.

NAME of package: apriltag_detector

# The source is here:
https://github.com/ros-misc-utilities/apriltag_detector

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

**Review by wjwwood — APPROVED** [review]
Sounds ok to me, waiting on another review to confirm the naming is ok with others too.

**Review by mjcarroll — APPROVED** [review]
Makes sense to me.

File: rolling/distribution.yaml
Comment (mjcarroll) [inline rolling/distribution.yaml:361]: Note that your next step here will be to get an entry in the ros2-gbp org: https://github.com/ros2-gbp/ros2-gbp-github-org

**Review by clalancette — APPROVED** [review]

**Pull request merged**
